src/mlpack/methods/ann/activation_functions/swish_function.hpp

```diff
@@ -70,26 +70,32 @@ class SwishFunction
   /**
    * Computes the first derivative of the swish function.
    *
-   * @param y Input data.
+   * @param x Input activation.
+   * @param y Result of Fn(x).
    * @return f'(x)
    */
-  static double Deriv(const double y)
+  static double Deriv(const double x, const double y)
   {
-    return y / (1 + std::exp(-y)) + (1 - y / (1 + std::exp(-y))) /
-                                             (1 + std::exp(-y));
+    double sigmoid = y / x; // save an exp
+    return x == 0 ? 0.5 : sigmoid * (1.0 + x * (1.0 - sigmoid));
```

Comment:
This is the same as the SILU function; I think here we should remove one of the two also.

Reply:
yup, or see me comment on the softplus.  do we want some sort of alias (typedef) since different people may use one or the other?

Reply:
probably as a separate PR though either way, just so that change isn't buried in all of this.

Reply:
Yeah, definitely, a typedef with one underlying implementation is the way to go for sure.  I agree with doing it as a separate PR. :+1: